main/lsp/LSPTypechecker.cc

```diff
@@ -43,6 +43,8 @@ LSPTypechecker::LSPTypechecker(std::shared_ptr<const LSPConfiguration> config)
 
 void LSPTypechecker::initialize(LSPFileUpdates updates, WorkerPool &workers) {
     ENFORCE(this_thread::get_id() == typecheckerThreadId, "Typechecker can only be used from the typechecker thread.");
+    ENFORCE(!this->initialized);
```

Comment:
I beefed up some enforces to ensure that we never try to initialize twice, and that we never try to use an uninitialized global state.

Reply:
That's it.